lib/tasks/development.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+namespace :dev do
+  require_relative 'build_utils'
+  require 'yaml'
+  require 'bcrypt'
+  include BuildUtils
+
+  def dev_container_name
+    'ood-dev' || ENV['OOD_DEV_CONTAINER_NAME'].to_s
+  end
+
+  def init_ood_portal
+    file = "#{config_directory}/ood_portal.yml"
+    return if File.exist?(file)
+
+    File.open(file, File::WRONLY|File::CREAT|File::EXCL) do |f|
+      f.write({
+        'servername': 'localhost',
+        'port': 8080,
+        'listen_addr_port': 8080,
+        'oidc_remote_user_claim': 'email',
+        'dex': {
+          'connectors': [{
+            'type': 'mockCallback',
+            'id': 'mock',
+            'name': 'Mock'
+          }]
+        }
+      }.to_yaml)
+    end
+  end
+
+  def init_ctr_user
+    file = "#{config_directory}/static_user.yml"
+    return if File.exist?(file)
+
+    require 'io/console'
+    puts 'Enter password:'
+    plain_password = $stdin.noecho(&:gets).chomp
+    bcrypted = BCrypt::Password.create(plain_password)
+
+    content = <<~CONTENT
```

Comment:
I wonder if we should make it possible to define a Dex user via ood-portal-generator so you don't have to write out a second file for the Dex local user.  I believe right now we hardcode in a default local user, which I think was just not thinking people or even developers might want to choose a different user besides the default we provide

Reply:
I think we need a way to provide development containers with such a feature. Enabling it in ood-portal-generator - I'm not sure. I've been told someone kept that `ood` user around during the basic auth days and it went sideways for them.  I mean they accidentally kept it and someone else started using it.

I think there could be a discourse topic where someone wanted to do something similar for a proof of concept deployment. So maybe we should provide a way to generate a single user with a new password.  For proof of concept deployments and this container.

Reply:
I think the biggest argument for using ood-portal-generator for generating the user is we already manage the Dex YAML config with ood-portal-generator so having ood-portal-generator make the config and then modifying it later without ood-portal-generator seems messy and will require subsequent executions of ood-portal-generator to require the --force flag.  I think taking the work here for generating a bcrypt password and supporting something similar with ood-portal-generator shouldn't be too big a change and pretty easy to maintain backwards compatibility.